File: benchmark/prime_sieve/sieve.py
def main():
    limit = 10000000
    size = limit + 1
    is_prime = bytearray([1]) * size
    is_prime[0] = 0
    is_prime[1] = 0

    p = 2
    while p * p <= limit:
        if is_prime[p]:
            for j in range(p * p, limit + 1, p):
                is_prime[j] = 0
        p += 1

    count = sum(1 for x in is_prime if x)
    print(f"Primes up to {limit}: {count}")


if __name__ == "__main__":
    main()

File: benchmark/prime_sieve/sieve.go
package main

import "fmt"

func main() {
	limit := 10000000
	size := limit + 1
	isPrime := make([]byte, size)

	for i := 0; i < size; i++ {
		isPrime[i] = 1
	}

	isPrime[0] = 0
	isPrime[1] = 0

	for p := 2; p*p <= limit; p++ {
		if isPrime[p] != 0 {
			for j := p * p; j <= limit; j += p {
				isPrime[j] = 0
			}
		}
	}

	count := 0
	for i := 0; i <= limit; i++ {
		if isPrime[i] != 0 {
			count++
		}
	}

	fmt.Printf("Primes up to %d: %d\n", limit, count)
}
